File: VB/WebSite/App_Code/BLL/CarsXtraSchedulingProxy.vb
Imports System
Imports System.Data.SqlClient
Imports System.ComponentModel
Imports CarsXtraSchedulingTableAdapters

<DataObjectAttribute()> _
Public NotInheritable Class CarsXtraSchedulingProxy

    Private Sub New()
    End Sub

    Private Shared carSchedulingTableAdapter As CarSchedulingTableAdapter
    Private Shared carTableAdapter As CarsTableAdapter

    Private Shared lastInsertedAppointmentId_Renamed As Integer

    Public Shared ReadOnly Property LastInsertedAppointmentId() As Integer
        Get
            Return lastInsertedAppointmentId_Renamed
        End Get
    End Property

    Shared Sub New()
        carSchedulingTableAdapter = New CarSchedulingTableAdapter()
        carTableAdapter = New CarsTableAdapter()
    End Sub

    <DataObjectMethodAttribute(DataObjectMethodType.Select)> _
    Public Shared Function GetAppointments() As CarsXtraScheduling.CarSchedulingDataTable
        Return carSchedulingTableAdapter.GetData()
    End Function

    <DataObjectMethodAttribute(DataObjectMethodType.Select)> _
    Public Shared Function GetResources() As CarsXtraScheduling.CarsDataTable
        Return carTableAdapter.GetData()
    End Function

    <DataObjectMethodAttribute(DataObjectMethodType.Insert)> _
    Public Shared Function InsertAppointment(ByVal CarId As String, ByVal Status? As Integer, ByVal Subject As String, ByVal Description As String, ByVal Label? As Integer, ByVal StartTime? As Date, ByVal EndTime? As Date, ByVal Location As String, ByVal AllDay As Boolean, ByVal EventType? As Integer, ByVal RecurrenceInfo As String, ByVal ReminderInfo As String, ByVal Price? As Decimal) As Boolean

        Dim rowsAffected As Integer = carSchedulingTableAdapter.Insert(CarId, Status, Subject, Description, Label, StartTime, EndTime, Location, AllDay, EventType, RecurrenceInfo, ReminderInfo, Price)

        UpdateLastInsertedAppointmentId()

        Return rowsAffected = 1
    End Function

    <DataObjectMethodAttribute(DataObjectMethodType.Update)> _
    Public Shared Function UpdateAppointment(ByVal CarId As String, ByVal Status? As Integer, ByVal Subject As String, ByVal Description As String, ByVal Label? As Integer, ByVal StartTime? As Date, ByVal EndTime? As Date, ByVal Location As String, ByVal AllDay As Boolean, ByVal EventType? As Integer, ByVal RecurrenceInfo As String, ByVal ReminderInfo As String, ByVal Price? As Decimal, ByVal ID As Integer) As Boolean

        Dim rowsAffected As Integer = carSchedulingTableAdapter.Update(CarId, Status, Subject, Description, Label, StartTime, EndTime, Location, AllDay, EventType, RecurrenceInfo, ReminderInfo, Price, ID)

        Return rowsAffected = 1
    End Function

    <DataObjectMethodAttribute(DataObjectMethodType.Delete)> _
    Public Shared Function DeleteAppointment(ByVal ID As Integer) As Boolean
        Dim rowsAffected As Integer = carSchedulingTableAdapter.Delete(ID)

        Return rowsAffected = 1
    End Function

    Private Shared Sub UpdateLastInsertedAppointmentId()
        carSchedulingTableAdapter.Connection.Open()
        Using command As New SqlCommand("SELECT IDENT_CURRENT('CarScheduling')", carSchedulingTableAdapter.Connection)
            lastInsertedAppointmentId_Renamed = Convert.ToInt32(command.ExecuteScalar())
        End Using
        carSchedulingTableAdapter.Connection.Close()
    End Sub

End Class

File: CS/WebSite/App_Code/CarsXtraSchedulingProxy.cs
using System;
using System.Data.SqlClient;
using System.ComponentModel;
using CarsXtraSchedulingTableAdapters;

[DataObjectAttribute()]
public static class CarsXtraSchedulingProxy {
    private static CarSchedulingTableAdapter carSchedulingTableAdapter;
    private static CarsTableAdapter carTableAdapter;
    private static int lastInsertedAppointmentId;

    public static int LastInsertedAppointmentId { get { return lastInsertedAppointmentId; } }

    static CarsXtraSchedulingProxy() {
        carSchedulingTableAdapter = new CarSchedulingTableAdapter();
        carTableAdapter = new CarsTableAdapter();
    }

    [DataObjectMethodAttribute(DataObjectMethodType.Select)]
    public static CarsXtraScheduling.CarSchedulingDataTable GetAppointments() {
        return carSchedulingTableAdapter.GetData();
    }

    [DataObjectMethodAttribute(DataObjectMethodType.Select)]
    public static CarsXtraScheduling.CarsDataTable GetResources() {
        return carTableAdapter.GetData();
    }

    [DataObjectMethodAttribute(DataObjectMethodType.Insert)]
    public static bool InsertAppointment(string CarId,
                                         int? Status,
                                         string Subject,
                                         string Description,
                                         int? Label,
                                         DateTime? StartTime,
                                         DateTime? EndTime,
                                         string Location,
                                         bool AllDay,
                                         int? EventType,
                                         string RecurrenceInfo,
                                         string ReminderInfo,
                                         decimal? Price) {

        int rowsAffected = carSchedulingTableAdapter.Insert(CarId,
                                                            Status,
                                                            Subject,
                                                            Description,
                                                            Label,
                                                            StartTime,
                                                            EndTime,
                                                            Location,
                                                            AllDay,
                                                            EventType,
                                                            RecurrenceInfo,
                                                            ReminderInfo,
                                                            Price);

        UpdateLastInsertedAppointmentId();

        return rowsAffected == 1;
    }

    [DataObjectMethodAttribute(DataObjectMethodType.Update)]
    public static bool UpdateAppointment(string CarId,
                                         int? Status,
                                         string Subject,
                                         string Description,
                                         int? Label,
                                         DateTime? StartTime,
                                         DateTime? EndTime,
                                         string Location,
                                         bool AllDay,
                                         int? EventType,
                                         string RecurrenceInfo,
                                         string ReminderInfo,
                                         decimal? Price,
                                         int ID) {

        int rowsAffected = carSchedulingTableAdapter.Update(CarId,
                                                            Status,
                                                            Subject,
                                                            Description,
                                                            Label,
                                                            StartTime,
                                                            EndTime,
                                                            Location,
                                                            AllDay,
                                                            EventType,
                                                            RecurrenceInfo,
                                                            ReminderInfo,
                                                            Price,
                                                            ID);

        return rowsAffected == 1;
    }

    [DataObjectMethodAttribute(DataObjectMethodType.Delete)]
    public static bool DeleteAppointment(int ID) {
        int rowsAffected = carSchedulingTableAdapter.Delete(ID);

        return rowsAffected == 1;
    }

    private static void UpdateLastInsertedAppointmentId() {
        carSchedulingTableAdapter.Connection.Open();
        using (SqlCommand command = new SqlCommand("SELECT IDENT_CURRENT('CarScheduling')", carSchedulingTableAdapter.Connection)) {
            lastInsertedAppointmentId = Convert.ToInt32(command.ExecuteScalar());
        }
        carSchedulingTableAdapter.Connection.Close();
    }

}